extensions/markdown-language-features/src/features/workspaceSymbolProvider.ts

```diff
@@ -0,0 +1,50 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { workspace, WorkspaceSymbolProvider, SymbolInformation, TextDocument } from 'vscode';
+import { isMarkdownFile } from '../util/file';
+import MDDocumentSymbolProvider from './documentSymbolProvider';
+import { Dictionary, flatMap } from 'lodash';
+
+export default class MarkdownWorkspaceSymbolProvider implements WorkspaceSymbolProvider {
+	private symbolProvider: MDDocumentSymbolProvider;
+	private symbolCache: Dictionary<SymbolInformation[]> = {};
+
+	public constructor(symbolProvider: MDDocumentSymbolProvider) {
+		this.symbolProvider = symbolProvider;
+		this.populateSymbolCache();
```

Comment:
This should be done lazily when workspace symbols are first requested

Reply:
done..